packages/field/src/FormatMixin.js

```diff
@@ -171,9 +171,26 @@ export const FormatMixin = dedupeMixin(
       }
 
       __callParser(value = this.formattedValue) {
-        let result;
-        if (typeof value === 'string') {
-          result = this.parser(value, this.formatOptions);
+        if (value === '') {
+          // Ideally, modelValue should be undefined for empty strings.
+          // For backwards compatibility we return an empty string.
+          return '';
+        }
+
+        if (typeof value !== 'string') {
```

Comment:
Scenario? When could this happen? I thought native inputs values were always of type String.

Reply:
More exceptional cases (extending layers) that call `__callParser` externally. Although chances are slim and it's mainly there for backwards compatibility.

Reply:
In the next release it should become 
`if (typeof value !== 'string' || value = '') {
  return undefined;
}`

And I agree that the typeof check doesn't make sense if you expect a string always.
I think it also covers for edge cases (this.value is delegated to this.inputElement.value but inputElement not available yet). Defensive code that's there for backw. compat.